packages/datepicker-react/src/DatePicker.test.tsx

```diff
@@ -407,6 +407,25 @@ describe("Datepicker", () => {
     });
 });
 
+describe("should call onBlur when input value is a string", () => {
+    const onBlur = jest.fn();
+    render(
+        <div>
+            <DatePicker onBlur={onBlur} />
+            <button>Click</button>
+        </div>,
+    );
+
+    const inputElement = screen.getByLabelText("Velg dato");
+
+    act(() => {
```

Comment:
Hvis du kan unngå å bruke act så tror jeg det er å foretrekke. Act er litt escape hatch når man ikke får det til på den "ordentlige" måten. 

Som regel kommer du unna ved å gjøre testen async og awaite et element er det etter state change har skjedd.

```
test("should not explode", async () => {
  const MyComponent = () => {
    const [isClicked, setIsClicked] = useState(false);
    return (
      <div>
        <DatePicker onBlur={onBlur} />
        <button onClick={() => setIsClicked(true)}>Click</button>
        {isClicked && <span data-testid="isclicked" />}
      </div>
    )
  }
  render(<MyComponent />);
       const inputElement = screen.getByLabelText("Velg dato"); 
       userEvent.click(inputElement);
       userEvent.type(inputElement, "1.januar");
       userEvent.click(screen.getByText("Click"));
  
       await screen.findByTestid("isclicked");
       expect(onBlur).toHaveBeenCalledTimes(1);
});
```

Btw, burde du ikke sjekke at verdien faktisk settes til inputfeltet, for denne testen ville vel ha passert tidligere også?
```

Reply:
> Hvis du kan unngå å bruke act så tror jeg det er å foretrekke. Act er litt escape hatch når man ikke får det til på den "ordentlige" måten.

Hmm.. Ser jo ut som de anbefaler å bruke den?

> Wrap any code rendering and triggering updates to your components into act() calls.
>
> Ensures that the behavior in your tests matches what happens in the browser more closely by executing pending useEffects before returning. This also reduces the amount of re-renders done.

Reply:
https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning

> Again, most of the time you shouldn't actually have to use act directly (thanks to React Testing Library's async utilities), but when you do, you'll know why it's needed and fixes the problem.

Reply:
Jeg har kommunisert med @frisol97 om denne PR og fikset andre punkter som alle har bidratt til. I tillegg, ser jeg at **act** ble brukt på mange linjer i DatePicker.test.tsx og jeg vet ikke mye om den. Kan vi merge det uten å fjerne **act**? Kanskje vi legger til en refaktorering oppgave som fjerner act fra alle test filer.